src/api/PluginManager.ts

```diff
@@ -43,28 +57,27 @@ export async function startPlugins() {
                 zip.closeEntry();
 
                 if (manifest.name in plugins) throw new Error(`Plugin ${manifest.name} already registered`);
-                if (!isPluginEnabled(manifest.name)) continue;
+                if (!isPluginEnabled(manifest.name)) {
+                    disabledPlugins[manifest.name] = manifest;
+                    continue;
+                }
 
-                zip.openEntry("index.js.bundle");
-                const pluginBuffer = zip.readEntry("binary");
-                zip.closeEntry();
+                const pluginBuffer = loadPluginBundle(zip);
 
                 const pluginClass = AliuHermes.run(file.name, pluginBuffer) as typeof Plugin;
                 try {
                     if (pluginClass.prototype instanceof Plugin) {
                         if (manifest.name !== pluginClass.name) throw new Error(`Plugin ${manifest.name} must export a class named ${manifest.name}`);
                         logger.info(`Loading Plugin ${manifest.name}...`);
-                        const plugin = plugins[manifest] = new pluginClass(manifest);
+                        const plugin = plugins[manifest.name] = new pluginClass(manifest);
                         plugin.start();
                     } else throw new Error(`Plugin ${manifest.name} does not export a valid Plugin`);
                 } catch (err) {
                     logger.error(`Failed while loading Plugin: ${manifest.name}\n`, err);
                 }
             } catch (err) {
                 logger.error(`Failed while loading Plugin ZIP: ${file.name}\n`, err);
-                zip.close();
-            } finally {
-                await zip.close();
+                zip?.close();
```

Comment:
this should be in a finally, not catch

Reply:
since the loadPluginBundle func closes the zip passed, its better off to just remove it right?